R/access-gsplot.R

```diff
@@ -113,31 +117,55 @@ lim <- function(object, side=NULL, axis = NULL, set.undefined=TRUE, if.null=c(0,
     
   }
   
-  lims <- lapply(side.names, function(x) {
+  lims <- lapply(all.side.names, function(x) {
     lim <- object[[x]]$lim
     if (object[[x]]$reverse){
       lim <- rev(lim)
     }
     return(lim)
   })
-  names(lims) <- side.names
-  if (!is.null(side) && length(side)==1){
-    lims <- lims[[1]]
-    if (set.undefined && all(is.na(lims))){
-      lims <- lim(object, axis=as.axis(side))
-      sides <- as.side(names(lims)[sapply(lims, function(x) !any(is.na(x)))])
-      closest.side <- sides[which.min(abs(side-sides))][1]
-      if (is.null(closest.side)){
-        lims <- NULL
-      } else {
-        lims <- lims[[as.side_name(closest.side)]]  
+  names(lims) <- all.side.names
+  
+  if(set.undefined){  
```

Comment:
do you need to update the tests for limits or do we have decent coverage of these new changes?